### DISCUSSION TITLE
Mesmerizer Delivery!! THANK YOU

### DISCUSSION THREAD

**Discussion by stretch911** [PlummersSoftwareLLC/NightDriverStrip#412] [General]
very excited to get a box from dave's garage.
Very easy setup....very very cool display effect...however having issue with picaso...see below. this is consistent. reboots, starts picaso, then freaks out...and reboots.

https://photos.app.goo.gl/E1ZUf9soYtCmCsF86

**Comment by robertlipe** [comment]
That's great! (that you got it)
That's terrible! (that it works for, like 8 seconds)

That's the Picaso3in1 affect. It goes in three stages - 1 is kind of a
light saber thing, 2 is the stick thing, and 3 is circles.

If you disable Picaso via the web interface, does any other effect do this?
I'm wondering if it's Piaso or if it's just the first thing that's running.
Alternately, if you select the next effect either via the web or via the
remote, does the next one crash?

Is this with code built today? Trunk or one of my branches?


Clearly, crashing is bad, but I'm trying to figure out what makes this
special.

RJL

RJL

**Reply by stretch911** [reply]
dude...its bada%% no worries. it just busts a boundary or something and chip freaks.

**Comment by robertlipe** [comment]
For now, let's assume this is my fault somehow. Picaso is "mine", but I
didn't think it was turned on in the trunk.

Was this out of the box or was it flashed with code that you built?
Has anyone else unboxed theirs? Same problem?

**Comment by stretch911** [comment]
pre-programmed board...just plugged it in and letting it run. lots of great stuff..love the fire...and wow..hard ot narrow it down

**Reply by stretch911** [reply]
plasma, powder, lava lamp all very cool...not to say they all aren't awesome

**Reply by stretch911** [reply]
liquid...nova...man

**Reply by stretch911** [reply]
circuit is very MATRIXY! love it

**Comment by robertlipe** [comment]
Well, several of those are "mine" and haven't officially landed yet, so
thank you.

We recently made liquid run the other direction:
https://github.com/robertlipe/NightDriverStrip/issues/2
I busted Lava into several pieces recently in
https://github.com/robertlipe/NightDriverStrip/issues/4
I broke Picasasso into three pieces yesterday at
https://github.com/robertlipe/NightDriverStrip/issues/6

Is it always Picasso that's crashing?

**Comment by robertlipe** [comment]
Wait, if it's always Picasso and it's always the second screen (sticks, not
"capes" and not circles) is it always when it bonks into the right edge of
the screen?

If so, it's  https://github.com/robertlipe/NightDriverStrip/issues/21
and I fixed that a while back.

If that's it, and you're comfortable building firmware and futzing with
Github (and I thnk that was a requirement...) I can coach you into building
a tree that has all my shiny new effects AND has that crash fixed.

Or you can build from the official tree and it'll be fixed just because you
don't have that effect until
https://github.com/PlummersSoftwareLLC/NightDriverStrip/pull/398
is merged ... and I'm doing "final" review of that in another window right
now, but I don't know the review and integration process on that. It's ...
big.

RJL

**Comment by smilima** [comment]
@davepl, I also want to thank you for the Mesmerizer package!  I spent weeks trying to figure out why it would not connect to my network, although every other wireless device worked perfectly fine.  After many debugging efforts using every tool I know about including Wireshark (I remember when it was called Ethereal...) I eventually just pushed the button on the back of the router to reset to factory defaults.  It then started working...   Why this was the case, I don't understand although I have over 20 years of experience in networking.  I will leave that up to the deity of your choice.

Anyway, I am not bothered by this because the time invested to understand how the code works was worth it, and I didn't break any hardware (although I was worried the ESP was faulty).  

As a hardware/software engineer, I always tell the young engineers I mentor "If you aren't breaking something, you aren't doing anything!"  However, I always caveat this with don't do things out of pure desperation or frustration. This results in breaking more things than you should and sends out a red flag.  There are plenty of us willing to help out if you aren't understanding.  I hope we can adopt this philosophy within this community.

**Comment by robertlipe** [comment]
>
> (although I was worried the ESP was faulty).
>
Tiny devices with no screen or keyboard, but 73 dangling wires connected to
nearby *thingies* naturally arouse that suspicion.

> However, I always caveat this with don't do things out of pure desperation
> or frustration. This results in breaking more things than you should and
> sends out a red flag.
>
There's a fairly narrow band of prescribed sacrificial hardware.Break one
and that's just business. Break three and it's going on your permanent
record.

> There are plenty of us willing to help out if you aren't understanding. I
> hope we can adopt this philosophy within this community.
>

So far, we seem to have a pretty good record going, but I think this week
or so will be the first "big" wave of onboarding. This group should be
developer-heavy, and all on identical hardware kits so we should be braced
more for  "how do I flash firmware?", "how can I configure a debugger?" and
"how do I conditionally add files to the build?" style of helping each
other out than "I built my own board that doesn't work" and "can I power a
mile of LEDs from a phone charger from my Nokia?" style questions.

This should be the wave of helping the helpers get started. I usually enjoy
this phase of projects and will be ready to spend time reproducing and
helping others.

(I have a workaround and fix ready for Stretch's Picasso issue when he's
ready. Let's see if others are also affected....)

Welcome to all the new faces and names!

RJ
